docs/glossary.md

```diff
@@ -0,0 +1,258 @@
+# AR.IO Node Glossary
+
+This glossary defines key terms and concepts used throughout the AR.IO Node
+codebase. Terms are organized by category and alphabetically within each
+section.
+
+## Core Arweave Concepts
+
+**Block** - The fundamental unit of the Arweave blockchain containing a set of
+transactions. Each block has a unique height (sequential number) and hash
+identifier.
+
+**Chunk** - A 256KB segment of transaction data. Large transactions are split
```

Comment:
Is it commonplace to use KB for KiB around the codebase?

Reply:
Nah, this is a Claudism that keeps slipping in.